plugins/polio/tasks/api/refresh_lqas_im_data.py

```diff
@@ -0,0 +1,145 @@
+        res["config"] = config  # is this safe?
+        res["id_field"] = id_field
+        return res
+
+
+class CustomTaskSearchFilterBackend(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        search = request.query_params.get("search")
+        if search:
+            query = (
+                Q(name__icontains=search)
+                | Q(status__icontains=search)
+                | Q(launcher__first_name__icontains=search)
+                | Q(launcher__username__icontains=search)
+                | Q(launcher__last_name__icontains=search)
+            )
+            return queryset.filter(query)
+
+        return queryset
+
+
+class RefreshLQASIMDataViewset(ExternalTaskModelViewSet):
+    permission_classes = [permissions.IsAuthenticated, HasPermission(permission.POLIO, permission.POLIO_CONFIG)]  # type: ignore
+    http_method_names = ["get", "post", "patch"]
+    model = Task
+    filter_backends = [
+        filters.OrderingFilter,
+        CustomTaskSearchFilterBackend,
+        DjangoFilterBackend,
+    ]
+    filterset_fields = {"created_at": ["gte"], "ended_at": ["exact"], "started_at": ["gte"], "status": ["exact"]}
+    ordering_fields = ["created_at", "ended_at", "name", "started_at", "status"]
+    ordering = ["-started_at"]
+    # defaulting to LQAS
+    task_name = LQAS_TASK_NAME
+    config_slug = LQAS_CONFIG_SLUG
+
+    def get_serializer_class(self):
+        if self.request.method == "POST":
+            return RefreshLQASIMDataPostSerializer
+        return ExternalTaskSerializer
+
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        context.update({"config_slug": self.config_slug})
+        return context
+
+    def get_queryset(self):
+        user = self.request.user
+        account = user.iaso_profile.account
+        queryset = Task.objects.filter(account=account).filter(external=True)
+        authorized_countries = user.iaso_profile.org_units.filter(org_unit_type_id__category="COUNTRY")
+        if authorized_countries.count() > 0:
+            authorized_names = [f"{self.task_name}-{id}" for id in authorized_countries]
+            queryset = queryset.filter(name__in=authorized_names)
+        return queryset
+
+    @action(detail=False, methods=["get"], serializer_class=TaskSerializer)
+    def last_run_for_country(self, request):
+        serializer = RefreshLQASIMDataGetSerializer(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        country_id = request.query_params.get("country_id", None)
+        status_query = Q(status=SUCCESS) | Q(status=RUNNING) | Q(status=ERRORED)
+        queryset = self.get_queryset().filter(status_query).exclude(started_at__isnull=True)
+        # The filter is based on how the task name is generated by ExternalTaskPostSerializer
+        query = (
+            Q(name=self.config_slug) | Q(name=f"{self.config_slug}-{country_id}")
```

Comment:
It feels like these names of tasks should be generated in a function reused in multiple places and not regenerated (with risks of differences) in multiple place. Not blocking